[CI] RecoverySourceHandlerTests.testSendSnapshotSendsOps failures

This has just started failing on both `master` and `7.x`
There was also a failure as part of a recent PR build ([jenkins](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-1/3535/), [PR](https://github.com/elastic/elasticsearch/pull/59182), [Fixed?](https://github.com/elastic/elasticsearch/commit/865b6b5880ddca4b2b05cb9c28d343a528b92437))

**Build scan**: https://gradle-enterprise.elastic.co/s/wygtbrni3cjho

**Repro line**:
```
./gradlew ':server:test' \
  --tests "org.elasticsearch.indices.recovery.RecoverySourceHandlerTests.testSendSnapshotSendsOps" \
  -Dtests.seed=2ED9D667ADE123CA -Dtests.security.manager=true \
  -Dtests.jvm.argline="-XX:-UseConcMarkSweepGC -XX:-UseSerialGC -XX:+UseG1GC" \
  -Dtests.locale=mt-MT -Dtests.timezone=Antarctica/South_Pole -Druntime.java=11
```

**Reproduces locally?**: No

**Applicable branches**: master, 7.x

**Failure history**:
https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-6M,mode:quick,to:now))&_a=(columns:!(branch,build-id,message),filters:!(),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:%22testSendSnapshotSendsOps%22),sort:!(time,desc))

**Failure excerpt**:

> java.lang.AssertionError: 	
Expected: <9220044514112065136L>	
     but: was <9221782717110317574L>	
	at __randomizedtesting.SeedInfo.seed([2ED9D667ADE123CA:69BA901B76F84597]:0)	
	at org.elasticsearch.indices.recovery.RecoverySourceHandlerTests.testSendSnapshotSendsOps(RecoverySourceHandlerTests.java:268)

--- a/server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java
+++ b/server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java
@@ -246,7 +246,9 @@ public class RecoverySourceHandlerTests extends ESTestCase {
             public void indexTranslogOperations(List<Translog.Operation> operations, int totalTranslogOps, long timestamp, long msu,
                                                 RetentionLeases retentionLeases, long mappingVersion, ActionListener<Long> listener) {
                 shippedOps.addAll(operations);
-                checkpointOnTarget.set(randomLongBetween(checkpointOnTarget.get(), Long.MAX_VALUE));
+                if (randomBoolean()) {
+                    checkpointOnTarget.addAndGet(between(1, 20));
+                }
                 listener.onResponse(checkpointOnTarget.get());
             }
         };

